engine/src/flutter/shell/platform/embedder/embedder.cc

```diff
@@ -2416,6 +2416,10 @@ FlutterEngineResult FlutterEngineInitialize(size_t version,
         "Could not infer the Flutter project to run from given arguments.");
   }
 
+  if (args->engine_id != 0) {
+    run_configuration.SetEngineId(args->engine_id);
```

Comment:
Should we call `run_configuration.SetEngineId` _before_ we check `run_configuration.IsValid`? This would ensure we generate the correct error if we ever decide to validate the run configuration's engine ID

Reply:
I'll move it.